roles/gpu/tasks/main.yml

```diff
@@ -0,0 +1,25 @@
+---
+- name: Check how many NVidia devices is up and running (might take some time)
+  ansible.builtin.command: 'nvidia-smi -L'
+  register: smi
+  when: gpu_count|default(0) >= 1
+  changed_when: false
+  failed_when: false
+
+- name: Check driver version
+  ansible.builtin.command: '/usr/sbin/modinfo nvidia'
+  register: modinfo
+  changed_when: false
+  failed_when: false
+  when: gpu_count|default(0) >= 1
+
+- name: Install GPU driver if not all GPU devices are present and working
+  ansible.builtin.include_tasks: driver.yml
+  when: gpu_count|default(0) >= 1 and
+        (( smi.stdout|default([])|lower|regex_findall('nvidia')|length != gpu_count ) or
+        gpu_driver_version not in modinfo.stdout|default("")|regex_search("version:.*"))
+
+- name: Configure user and services
+  ansible.builtin.include_tasks: user_services.yml
```

Comment:
`user_services.yml` does not exist in this PR and `gpu.yml` does exist but is no longer used anywhere as far as I can tell. Looks like one or more commits are missing...